server/gateway.go

```diff
@@ -1619,42 +1836,75 @@ func (s *Server) sendQueueSubToGateways(accName string, sub *subscription, isRem
 	}
 }
 
-func (s *Server) sendQueueUnsubToGateways(accName string, qsub *subscription, isRemote bool) {
-	// If this is a remote queue sub, need to remove from remote queue subs map
-	if isRemote {
-		s.gateway.Lock()
-		delete(s.gateway.rqs, string(qsub.sid))
-		s.gateway.Unlock()
+// This is invoked when a (queue) subscription is added/removed locally
+// or in our cluster. We use ref counting to know when to update
+// the inbound gateways.
+// <Invoked from client or route connection's readLoop or when such
+// connection is closed>
+func (s *Server) gatewayUpdateSubInterest(accName string, sub *subscription, change int32) {
+	var (
+		keya  [1024]byte
+		key   = keya[:0]
+		entry *sitally
+		isNew bool
+	)
+
+	s.gateway.pasi.Lock()
```

Comment:
I would make gateway -> gw but move pasi to be more descriptive ;)

Reply:
Was going the Derek way ;-). I am ashamed to say how many minutes I have spent trying to name this. If you have some suggestion, that would be very welcomed!

Reply:
lol - naming things is hard. That and cache invalidation..

Reply:
.. so any suggestion?